common/models/user.js

```diff
@@ -668,6 +668,26 @@ module.exports = function(User) {
       next();
     });
 
+    // Delete old sessions once email is updated
+    UserModel.observe('before save', function(ctx, next) {
+      if (ctx.instance) {
```

Comment:
`updateAttributes` and `upsert` don't fill `ctx.instance`, they provide `ctx.data` instead.